pkg/agent/agent_windows.go

```diff
@@ -111,6 +117,104 @@ func (i *Initializer) prepareHNSNetworkAndOVSExtension() error {
 	return util.PrepareHNSNetwork(subnetCIDR, i.nodeConfig.NodeTransportIPv4Addr, adapter, i.nodeConfig.UplinkNetConfig.Gateway, dnsServers, i.nodeConfig.UplinkNetConfig.Routes, i.ovsBridge)
 }
 
+func (i *Initializer) prepareVmNetworkAndOVSExtension() error {
+	var eNodeIface *net.Interface
+	en, err := i.crdClient.CrdV1alpha1().ExternalNodes(i.namespace).Get(context.TODO(), i.GetNodeConfig().Name, metav1.GetOptions{})
+	if err != nil {
+		klog.InfoS("Failed to get ExternalNode", "Name", i.GetNodeConfig().Name)
+		return err
+	}
+	for _, intF := range en.Spec.Interfaces {
```

Comment:
By now, we only support one interface in `ExternalNode`, so here we can use `en.Spec.Interfaces[0]` directly.

Secondly, en is already got in function `initVMLocalConfig`, can we get the uplink configuration s in that function on Windows? Then we don't need a second remote get here.

Reply:
Do you mean to cache interface name (uplink name) into UplinkNetConfig in the initVMLocalConfig?
Then in that case, we will be caching it for both Windows and Linux.

Reply:
Yes, I mean we can find the expected uplink name in initVMLocalConfig, but we can have the logic only on Windows. On Linux, we don't care the uplinkNetAdapter, right?